python/lsst/pipe/base/configOverrides.py

```diff
@@ -125,35 +227,77 @@ def applyTo(self, config):
         ------
         `Exception` is raised if operations on configuration object fail.
         """
+        # Look up a stack of variables people may be useing when setting
+        # configs. Create a dictionary that will be used akin to a namespace
+        # for the duration of this function.
```

Comment:
This is a cool feature, though I'm not sure I understand how it is going to work or whether it's going to pick some names unintentionally from higher stack levels. I think some documentation and rationale would be useful (in user-facing docs).

Reply:
This feature exists basically to support (though it has other nice side effects) a few configuration level syntax being introduced on another ticket. The idea is something like this:

```python
label: TaskA
  class: some.PipelineTask
  config:
    action: A_Functor_Like_Action
    action.threshold: 17
    python: from action_library import A_Functor_Like_Action
```
This code would also allow variables to be found outside of an explicit import in certain cases. I was on the wall about letting people do imports prior to running `toExpandedPipeline` (where this code is triggered) and that is why the frame lookup is there. But like you say there might be unintended side effects, and so I think I will fall back to only allowing variable lookups in the python block, or in the module where the config was defined. This would still allow someone to import several standard actions that people might want to use, while likely not being too inclusive. We could always add more later if we choose to.

As far as documentation goes, I agree, and it will land either in Pipeline documentation or documentation on the new actions stuff when it comes out.